edb/server/server.py

```diff
@@ -527,6 +539,34 @@ async def reload_sys_config(self):
         self._dbindex.update_sys_config(cfg)
         self._reinit_idle_gc_collector()
 
+    def schedule_reported_config_if_needed(self, setting_name):
+        setting = self._config_settings[setting_name]
+        if setting.report:
+            self.create_task(
+                self.load_reported_config(), interruptable=True)
+
+    def get_report_config_data(self) -> bytes:
+        return self._report_config_data
+
+    async def load_reported_config(self):
+        syscon = await self._acquire_sys_pgcon()
+        try:
+            data = await syscon.parse_execute_extract_single_data_frame(
+                self.get_sys_query('report_configs'),
+                b'__report_configs',
+                dbver=0, use_prep_stmt=True, args=(),
+            )
+            self._report_config_data = (
+                struct.pack('!L', len(self._report_config_typedesc)) +
+                self._report_config_typedesc +
+                data
+            )
```

Comment:
Even though it's very unlikely, but should we handle the case when `data` is `None`? Or perhaps the underlying query should raise an error instead of returning `None`?

Reply:
> but should we handle the case when data is None?

`self._report_config_typedesc` would either be set to a non-None value or not set at all (prompting an AttributeError). I think we're fine here.

Reply:
Oh sorry I mean `data` returned from `syscon.parse_execute_extract_single_data_frame(...)`